### PR TITLE
[ML] Change JobSchedule Type Hints -> Schedule Type Hints

### PR DESCRIPTION
**Pull request by needuv**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py [hunk 1]
```diff
@@ -161,14 +161,14 @@ def get(
     def begin_create_or_update(
         self,
         schedule,
```
Comment (annatisch) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py:163]: We seem to be missing a type hint
```suggestion
        schedule: Schedule,
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_schedule/schedule.py [hunk 2]
```diff
@@ -78,6 +79,19 @@ def _resolve_cls_and_type(cls, data, params_override):  # pylint: disable=unused
...
+        raise ScheduleException(
+            message=msg,
+            no_personal_data_message=msg,
+            target=ErrorTarget.SCHEDULE,
+            error_category=ErrorCategory.SYSTEM_ERROR,
+        )
+
     @property
     def create_job(self) -> None:
         module_logger.warning("create_job is not a valid property of %s", str(type(self)))
```
Comment (annatisch) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_schedule/schedule.py:97]: Can we add a `return None` here, just to be explicit?

Comment (needuv) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_schedule/schedule.py:97]: added

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py (hunk 1 above)
Comment (needuv) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py]: added. I also added type hints for some other arguments that were missing them

**Review by annatisch — APPROVED** [review]

**Review by luigiw — APPROVED** [review]

**Pull request merged**